compiler/modules/nback/number.bal

```diff
@@ -174,39 +183,51 @@ function buildConvertToFloat(llvm:Builder builder, Scaffold scaffold, bir:Conver
         buildConvertIntToFloat(builder, scaffold, val, insn);
         return;
     }
-    else if repr.base != BASE_REPR_TAGGED {
-        return scaffold.unimplementedErr("convert form decimal to float", insn.pos);
+    else if repr == REPR_DECIMAL {
```

Comment:
We can write this so it will continue to work with subtypes of DECIMAL.

Check for repr.base == BASE_REPR_TAGGED and the bitset is just DECIMAL.

Reply:
Changed

Reply:
> Lets add some test converting form unions types. eg `int|float` `int|float|decimal` `int|boolean`

Added more tests